src/EditorFeatures/Test2/Diagnostics/DiagnosticProviderTests.vb

```diff
@@ -379,17 +379,17 @@ Namespace Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics.UnitTests
                        x.Severity = y.Severity AndAlso
                        x.ProjectId = y.ProjectId AndAlso
                        x.DocumentId = y.DocumentId AndAlso
-                       Equals(x.DataLocation?.OriginalStartLine, y.DataLocation?.OriginalStartLine) AndAlso
-                       Equals(x.DataLocation?.OriginalStartColumn, y.DataLocation?.OriginalStartColumn)
+                       Equals(x.DataLocation.OriginalStartLine, y.DataLocation.OriginalStartLine) AndAlso
+                       Equals(x.DataLocation.OriginalStartColumn, y.DataLocation.OriginalStartColumn)
             End Function
 
             Public Overloads Function GetHashCode(obj As DiagnosticData) As Integer Implements IEqualityComparer(Of DiagnosticData).GetHashCode
                 Return Hash.Combine(obj.Id,
                        Hash.Combine(obj.Message,
                        Hash.Combine(obj.ProjectId,
                        Hash.Combine(obj.DocumentId,
-                       Hash.Combine(If(obj.DataLocation?.OriginalStartLine, 0),
-                       Hash.Combine(If(obj.DataLocation?.OriginalStartColumn, 0), obj.Severity))))))
```

Comment:
consumption side is now much more pleasant.  without needing `??` all over the place.